app/src/main/java/org/simple/clinic/signature/SignatureEffectHandler.kt

```diff
@@ -1,12 +1,24 @@
 package org.simple.clinic.signature
 
 import com.spotify.mobius.rx2.RxMobius
+import com.squareup.inject.assisted.Assisted
+import com.squareup.inject.assisted.AssistedInject
 import io.reactivex.ObservableTransformer
+import org.simple.clinic.util.scheduler.SchedulersProvider
 
-class SignatureEffectHandler {
+class SignatureEffectHandler @AssistedInject constructor(
+    @Assisted private val ui: SignatureUi,
```

Comment:
We should be passing `SignatureUiActions` over here. We use `SignatureUi` in UI renderer, any one of actions should be present in `SignatureUiActions` interface.